modules/sso.php

```diff
@@ -746,7 +865,42 @@ function allowed_redirect_hosts( $hosts ) {
 		return array_unique( $hosts );
 	}
 
-	function button( $args = array() ) {
+	/**
+	 * Builds the "Login to WordPress.com" button that is displayed on the login page as well as user profile page.
+	 *
+	 * @param  array   $args       An array of arguments to add to the SSO URL.
+	 * @param  boolean $is_primary Should the button have the `button-primary` class?
+	 * @return string              Returns the HTML markup for the button.
+	 */
+	function build_sso_button( $args = array(), $is_primary = false ) {
+		$url = $this->build_sso_button_url( $args );
+		$classes = $is_primary
+			? 'jetpack-sso button button-primary'
+			: 'jetpack-sso button';
+
+		return wp_kses(
```

Comment:
I think this could be greatly simplified by just using 
return `'<a href="'.esc_url( $url ).'" class="'.esc_attr($classes).'">'.esc_html__( 'Log in with WordPress.com', 'jetpack' ).'</a>';`